### ISSUE TITLE
Welding goggles/helmets issue

### ISSUE THREAD

**Issue by uazuaz** [d3athrow/vgstation13#11629]
#### Description of issue

welding goggles and helmets should limit your vision, but sometimes this doesn't happen at all, sometimes sight is limited for just a couple of seconds then there is no limit again.
![weld](https://cloud.githubusercontent.com/assets/6170726/18033424/400ceaf4-6d24-11e6-8817-991f77fe8586.jpg)
#### Difference between expected and actual behavior

having limited sight when wearing welding goggles and helmets;
sometimes there is no limited sight.
#### Steps to reproduce

wear welding goggles or helmet, then i'm not really sure, try to toggle it multiple times.
#### Length of time in which bug has been known to occur

noticed the first time a couple of days ago.
#### Server revision

6d7cb3a7af9ef0291b1f855a3b62321c93608c87 
#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

**Comment by PJB3005** [comment]
@gbasood

**Comment by Kurfursten** [comment]
The way this behaves reminds me a lot of when I fixed how earmuffs would only deafen you some of the time. What was happening was, the earmuffs would set your ears to 1 point of deaf if you didn't have deafness, then periodic healing would repair your ears after one second. So you'd constantly cut in and out.

I get the feeling this is something similar in life.

**Comment by gbasood** [comment]
~~literally
fixed this
I get the feeling someone 'fixed' something related to this and broke it all over again~~

**Comment by gbasood** [comment]
ah so nevermind that's the problem
It'll clear your fullscreen when it shouldn't sometimes

``` DM
if(istype(head, /obj/item/clothing/head/welding) || istype(head, /obj/item/clothing/head/helmet/space/unathi))
            var/obj/item/clothing/head/welding/O = head
            if(!O.up && tinted_weldhelh)
                overlay_fullscreen("tint", /obj/screen/fullscreen/impaired, 2)
                masked = 1
            else
                clear_fullscreen("tint")
        else
            clear_fullscreen("tint")

        if(!masked && istype(glasses, /obj/item/clothing/glasses/welding) && !istype(glasses, /obj/item/clothing/glasses/welding/superior))
            var/obj/item/clothing/glasses/welding/O = glasses
            if(!O.up && tinted_weldhelh)
                overlay_fullscreen("tint", /obj/screen/fullscreen/impaired, 2)
            else
                clear_fullscreen("tint")

```

will fix

**Comment by vonzippenstein** [comment]
While anyone is on this issue, a welding helmet when in the down position and equiped then lifted will erase your facial hair. But it can be avoided by equipping the helmet on while it's set to lifted.

**Comment by gbasood** [comment]
Should be fixed now with #11600 , please reply if it's not
The facial hair thing isn't me so I don't know how to fix, please make a separate report so it doesn't get lost
